components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -150,13 +175,93 @@ function AutomaticPointsTable:queryPlacements(tournaments)
 			local tournamentIndex = tournamentIndices[result.tournament]
 			local tournament = tournaments[tournamentIndex]
 
-			result.points = tonumber(result.extradata.prizepoints)
+			result.prizePoints = tonumber(result.extradata.prizepoints)
 			result.securedPoints = tonumber(result.extradata.securedpoints)
+			result.extradata = nil
 			table.insert(tournament.placements, result)
+
+			local participant = result.participant
+			local teamIndex = reverseAliasLookupTable[tournamentIndex][participant]
+			if teamIndex ~= nil then
+				teams[teamIndex].results[tournamentIndex] = result
+			end
 		end
 	)
 
-	return tournaments
+	return teams, tournaments
+end
+
+function AutomaticPointsTable:getPointsData(teams, tournaments)
+	local pointsData = {}
+	Table.iter.forEachIndexed(teams,
+		function(teamIndex, team)
+			local teamPointsData = {}
+			local totalPoints = 0
+			for tournamentIndex = 1, #tournaments do
+				local manualPoints = team.manualPoints[tournamentIndex]
+				local placement = team.results[tournamentIndex]
+				local tournamentTeamPointsData = self:getTournamentTeamPointsData(manualPoints, placement)
+				if Table.isNotEmpty(tournamentTeamPointsData) then
+					totalPoints = totalPoints + tournamentTeamPointsData.amount
+				end
+
+				local deduction = team.deductions[tournamentIndex]
+				local dataWithDeduction = self:getDeductionData(deduction, tournamentTeamPointsData)
+				if Table.isNotEmpty(dataWithDeduction.deduction) then
+					-- will only show the deductions column if there's atleast one team with
+					-- some deduction for a tournament
+					tournaments[tournamentIndex].deductionsVisible = true
+					totalPoints = totalPoints - dataWithDeduction.deduction.amount
+				end
+				teamPointsData[tournamentIndex] = tournamentTeamPointsData
+			end
+			teamPointsData.totalPoints = totalPoints
+			pointsData[teamIndex] = teamPointsData
+		end
+	)
+
+	return pointsData
+end
+
+function AutomaticPointsTable:getTournamentTeamPointsData(manualPoints, placement)
+	local pointsData = {}
+	-- manual points get highest priority
+	if manualPoints ~= nil then
+		pointsData = Table.mergeInto(pointsData, {

```

Comment:
The table is empty. Why are we merging? Just insert. Same comment for code below.

Reply:
Done